src/main.rs

```diff
@@ -56,9 +56,9 @@ fn main() {
 
     init_logger(&args);
 
-    let dir = args.arg_directory.map_or_else(determine_dir, |dir| PathBuf::from(dir));
+    let dir = args.arg_directory.clone().map_or_else(determine_dir, |dir| PathBuf::from(dir));
```

Comment:
You can get rid of the `clone()` if you change `walk_dir` to take a boolean instead of all of `MainArgs`. Alternatively you could change this to a `&Path` instead of a `PathBuf`.